src/AutoRest.CSharp/Common/AutoRest/Plugins/NewProjectScaffolding.cs

```diff
@@ -315,18 +315,18 @@ private string GetBrandedSrcCSProj()
                 IncludeOperationsSharedSource = true,
                 IncludeManagementSharedCode = Configuration.AzureArm ? true : null,
             };
-            // only branded library will add these shared code compilation lines
-            builder.CompileIncludes.Add(new("$(AzureCoreSharedSources)AzureResourceProviderNamespaceAttribute.cs", "Shared/Core"));
             if (_needAzureKeyAuth)
                 builder.CompileIncludes.Add(new("$(AzureCoreSharedSources)AzureKeyCredentialPolicy.cs", "Shared/Core"));
-            foreach (var packages in _brandedDependencyPackages)
+            if (!Configuration.AzureArm)
             {
-                builder.PackageReferences.Add(packages);
-            }
-            if (Configuration.AzureArm)
-            {
-                builder.PackageReferences.Add(new("Azure.ResourceManager"));
+                // only branded library will add these shared code compilation lines
+                builder.CompileIncludes.Add(new("$(AzureCoreSharedSources)AzureResourceProviderNamespaceAttribute.cs", "Shared/Core"));
```

Comment:
Not quite follow the change for this file.
Here, we are only adding it for DPG. And, we are adding it in [Directory.Build.targets](https://github.com/Azure/autorest.csharp/pull/4923/files#diff-50e91c82311ea26f2a73202525dfdf2b0a89c178ee666b2bf3ad4c84ac4c06dc) for mgmt.
Does it mean we need this for both DPG and Mgmt? We can just remain as it was, adding it for branded library.

Reply:
hello @live1206  It is consistent with azure-sdk-for-net. SDKs in azure-sdk-for-net will not add this in project file, and it add this in azure-sdk-for-net eng for all SDKs.
So in autorest.csharp, we will do the same settings. then the generated project file will not include this compile item. Otherwise, when put this SDK in azure-sdk-for-net, the build will fail.

Reply:
OK, then we need to regen update the test csproj files, CI is failing due to the reference order change as below
![image](https://github.com/user-attachments/assets/a8cf2541-cc13-4075-8391-dc0cc1a1b205)